cuml/forest_inference_demo.ipynb

```diff
@@ -0,0 +1,255 @@
+    "The forest inference library is used to load saved forest models of xgboost, lightgbm or protobuf and perform inference on them. It can be used to perform both classification and regression. In this notebook, we'll begin by fitting a model with XGBoost and saving it. We'll then load the saved model into FIL and use it to infer on new data.\n",
+    "\n",
+    "Forest inference library works in the same way with lightgbm and protobuf model as well.\n",
+    "\n",
+    "The model accepts both numpy arrays and cuDF dataframes. In order to convert your dataset to cudf format please read the cudf documentation on https://rapidsai.github.io/projects/cudf/en/latest/. \n",
+    "\n",
+    "For additional information on the forest inference library please refer to the documentation on https://rapidsai.github.io/projects/cuml/en/latest/index.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/nfs/saljain/miniconda3/envs/fil-test/lib/python3.7/site-packages/numba/cuda/envvars.py:16: NumbaDeprecationWarning: \n",
+      "Environment variables with the 'NUMBAPRO' prefix are deprecated, found use of NUMBAPRO_NVVM=/usr/local/cuda/nvvm/lib64/libnvvm.so.\n",
+      "\n",
+      "For more information visit http://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-numbapro-environment-variables\n",
+      "  warnings.warn(errors.NumbaDeprecationWarning(msg))\n",
+      "/home/nfs/saljain/miniconda3/envs/fil-test/lib/python3.7/site-packages/numba/cuda/envvars.py:16: NumbaDeprecationWarning: \n",
+      "Environment variables with the 'NUMBAPRO' prefix are deprecated, found use of NUMBAPRO_LIBDEVICE=/usr/local/cuda/nvvm/libdevice.\n",
+      "\n",
+      "For more information visit http://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-numbapro-environment-variables\n",
+      "  warnings.warn(errors.NumbaDeprecationWarning(msg))\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "import pytest\n",
+    "import os\n",
+    "\n",
+    "from cuml import ForestInference\n",
+    "from cuml.test.utils import array_equal\n",
+    "from cuml.utils.import_utils import has_xgboost\n",
+    "\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.metrics import accuracy_score\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "if has_xgboost():\n",
+    "    import xgboost as xgb\n",
+    "else:\n",
+    "    raise(\"Please install xgboost using the conda package,\"\n",
+    "          \" Use conda install -c conda-forge xgboost \"\n",
+    "          \"command to install xgboost\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### function that trains the xgboost model and performs prediction on it as well\n",
```

Comment:
Maybe instead of putting full sentences as headings, make:

```
## Train Helper Function

Define a simple function that trains the XGBoost model and performs prediction on it as well.

For additional information...
```